### PR TITLE
Date: Store & display ISO dates instead of datetimes

### LINKED ISSUE #11509: [v4] Date field always shows in CTM as 1 day prior
### Describe the bug

When setting the date field (unable to test datetime as it doesn't exist in this version yet), it always shows as 1 day previous but it correctly returned in the API response.

### Steps to reproduce the behavior

1. Add a date field to a content type
2. Create an entry and select a day
3. See that the CTM UI shows the day before
4. See that the REST Response is correct

### Expected behavior

Should show the correct day, most likely this is caused by the usage of the browsers timezone and is trying to render the date based on that location (although I've tested in the morning, afternoon, and evening all with the same result)

### Screenshots

https://www.loom.com/share/09f2ccd20db9462880efe78ac58b3c02

### Code snippets

N/A

### System

- Node.js version: v14.17.5
- NPM version: v6.14.14
- Strapi version: 4.0.0-beta.11
- Database: SQLite
- Operating system:  Linux Mint 20.2

### Additional context

N/A

### PR DESCRIPTION
**Pull request by HichamELBSI**
Signed-off-by: HichamELBSI <elabbassih@gmail.com>

### What does it do?

Send only the date for the date field type.

### How to test?

If you are using a production build or `yarn start` please make sure to rebuild the helper-plugin before:

```
cd /packages/core/helper-plugin
yarn build
```

### Related issue(s)/PR(s)

Fix #11509

Back end PR : https://github.com/strapi/strapi/pull/12234

### CODE REVIEW

**Review by ronronscelestes — APPROVED** [review]
LGTM!

**Comment by petersg83** [comment]
I'm not sure it's going to work properly with timezone. Can we discuss about it in live ?

**Comment by johan13** [comment]
> I'm not sure it's going to work properly with timezone. Can we discuss about it in live ?

I agree. See my last comment on #11509. If you e.g. select 2022-01-29 and is running in TZ UTC+1, then `toISOString()` returns `2022-01-28T23:00:00.000`. Splitting on `T` will result in `2022-01-28` which is still one day off.

**Review by soupette — DISMISSED** [review]
LGTM

**Comment by johan13** [comment]
@soupette, should this really be merged? I think the proposed solution will only work in timezones east of GMT.

File: packages/core/helper-plugin/lib/src/components/GenericInput/index.js [hunk 1]
```diff
@@ -166,7 +166,7 @@ const GenericInput = ({
           hint={hint}
           name={name}
           onChange={date => {
-            const formattedDate = date.toISOString();
+            const formattedDate = date.toISOString().split('T')[0];
```
Comment (johan13) [inline packages/core/helper-plugin/lib/src/components/GenericInput/index.js:169]: How about this instead?

```js
const tzAdjustedDate = new Date(date.getTime() - date.getTimezoneOffset() * 60000);
const formattedDate = tzAdjustedDate.toISOString().slice(0,10);
```

**Comment by soupette** [comment]
I am closing this PR. The backend should be responsible of managing dates therefore we should always send the following format `new Date().toISOString()`

File: packages/core/helper-plugin/lib/src/components/GenericInput/index.js (hunk 1 above)
Comment (HichamELBSI) [reply packages/core/helper-plugin/lib/src/components/GenericInput/index.js:169]: Unfortunately, this is not enough. On the display side, we create a Date. We have to change that too.

**Comment by petersg83** [comment]
Ahah I disagree @soupette, I think we were managing the type Date in the wrong way since the beginning. The right way I think would be to use the format `yyyy-MM-dd` in both the front end and the back end. I made the necessary change in the back (https://github.com/strapi/strapi/pull/12234).
It doesn't make sense to represent a Date (`yyyy-MM-dd`) with hours and timezones and it is really tricky to handle.

**Comment by gu-stav** [comment]
Re-opening this PR, since it has been discussed that the API will print a deprecation warning if something other than `YYYY-MM-DD` was passed through.

**Review by petersg83 — APPROVED** [review]
It works on my side, thanks :)

**Pull request merged**